### PR TITLE
impl Passowrd reset (aka Forgot password)

### PR DESCRIPTION
**Pull request by siddhantk232**
Add support for password reset by implementing some handlers. Specifically:

- `/forgot-password/`: Request a password reset email.
- `/set-password/`: Set a new password for a given user. Links in emails sent from `/forgot-password/` rely on this handler.

Also contains some minor refactoring.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ argon2 = "0.5"
 chrono = { version = "0.4", default-features = false, features = ["serde"] }
 validator = "0.18"
 cookie = "0.18"
-#ft-sdk = { version = "0.1.11", path = "../../ft-sdk/ft-sdk", features = ["sqlite-default", "auth-provider", "field-extractors"] }
-ft-sdk = { version = "0.1.14", features = ["sqlite-default", "auth-provider", "field-extractors"] }
+# ft-sdk = { path = "../ft-sdk/ft-sdk", features = ["sqlite-default", "auth-provider", "field-extractors"] }
+ft-sdk = { version = "0.1", features = ["sqlite-default", "auth-provider", "field-extractors"] }
```
Comment (amitu) [inline Cargo.toml:30]: I actually prefer the more precise version numbering of our crates.

File: email-auth-provider/src/handlers/set_password.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    validate_email_and_password(&email, &new_password, &new_password2)?;
+
+    let next = next.unwrap_or_else(|| "/".to_string());
+
+    let (user_id, data) = match ft_sdk::auth::provider::user_data_by_custom_attribute(
+        &mut conn,
+        email_auth::PROVIDER_ID,
+        email_auth::PASSWORD_RESET_CODE_KEY,
+        &code,
+    ) {
```
Comment (amitu) [inline email-auth-provider/src/handlers/set_password.rs:21]: This means we will have to index the reset code key field also. I think we can search by email as well, and email index will def exist, so we can save one index by changing query.

Comment (amitu) [reply email-auth-provider/src/handlers/set_password.rs:21]: We currently do not have indexes, which we will fix someday.

**Review by amitu — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (siddhantk232) [reply Cargo.toml]: Thanks. done in https://github.com/fastn-community/auth/pull/11/commits/7a284a269554b3e6d2dee5d0082b7cd98b15326e

File: email-auth-provider/src/handlers/set_password.rs (hunk 2 above)
Comment (siddhantk232) [reply email-auth-provider/src/handlers/set_password.rs:21]: We've similar code in one `conrfirm_email`. I'll leave this one as it is and we can come back to it when we've added an index on the email field.

**Pull request merged**